plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -165,6 +172,15 @@ export class DatabaseNotificationsStore implements NotificationsStore {
       q.where('user', user).orWhereNull('user');
     });
 
+    Object.keys(options.metadata ?? {}).forEach(key => {
+      query.whereJsonPath(
```

Comment:
To accept this, we must extend our test matrix for all supported databases in the `router.test.ts`, so far it is with `sqlite3` only.
We can mimic what other tests are doing with the `TestDatabases` from `@backstage/backend-test-utils`

Reply:
oh for sure, all database related code should be covered by TestDatabases (even if it's only for a subset of all possible databases). Do not use `DatabaseManager` directly in tests.

Reply:
But `DatabaseNotificationsStore.test.ts` [already uses that](https://github.com/backstage/backstage/blob/9b3399e06ec2ece812446cf0503eb443f65d07a4/plugins/notifications-backend/src/database/DatabaseNotificationsStore.test.ts#L153), right?

Reply:
Right, I was confused with the `router.test.ts` which can stay with one DB only as long as we do sufficient testing of the DatabseNotificationsStore.

@benjdlambert  can you please add more filtering cases for Json queries there? So far it's just full equality.

Reply:
@mareklibra I think you meant to ping @acierto 😅

Reply:
@mareklibra I'll close this PR as after following up discussion [here](https://github.com/backstage/backstage/pull/24562) it was decided to implement it slightly differently.

I'll raise soon another PR, and will include there better DB coverage